depends/libhdfs3/src/client/InputStreamImpl.cpp

```diff
@@ -626,6 +645,12 @@ int32_t InputStreamImpl::readInternal(char * buf, int32_t size) {
 
                 continue;
             }
+            std::string bufDecode;
+            if (fileStatus.isFileEncrypted()) {
+                /* Decrypt buffer if the file is encrypted. */
+                bufDecode = cryptoCodec->cipher_wrap(buf, size);
+                memcpy(buf, bufDecode.c_str(), size);
```

Comment:
must use `retval` instead of `size` here, because `readOneBlock()` doesn't always give "size" bytes.